### PR TITLE
[Play/Host] - Short Answer Response - Bug Fixes

### PR DESCRIPTION
**Pull request by drewjhart**
**Issue:**
This PR fixes some miscellaneous bugs based, for example if a student enter a plain number as an expression, how phase two answers are displayed after short answers in phase one, and preventing swiping when the quill pad is focused. 

_Note: this PR is a part of a multi-part PR for the Short Answer Responses feature. The full PR structure is below:
[[Play/Networking] - Short Answer Response Base #844](https://github.com/rightoneducation/righton-app/pull/844)
->[[Host] - Short Answer Response Base #845](https://github.com/rightoneducation/righton-app/pull/845)
->->[[Networking/Host] - Short Answer Response Updates to Backend #848](https://github.com/rightoneducation/righton-app/pull/848)
->->->[[Play] - Short Answer Response Scoring #849](https://github.com/rightoneducation/righton-app/pull/849)
->->->->[[Play/Host] - Game Flow #850](https://github.com/rightoneducation/righton-app/pull/850)
->->->->->[[Host] - Featured Mistakes #854](https://github.com/rightoneducation/righton-app/pull/854)
->->->->->->[[Play/Host] - Short Answer Formatting/Linting #855](https://github.com/rightoneducation/righton-app/pull/855)
->->->->->->->[[Play/Host] - Short Answer Response - Bug Fixes #856](https://github.com/rightoneducation/righton-app/pull/856)
->->->->->->->->[[Play/Host] - Short Answer - Helper Function Comments #857](https://github.com/rightoneducation/righton-app/pull/857)
->->->->->->->->->[[Host/Play/Networking] - Short Answer Response - Adjusts Answer Handling #858](https://github.com/rightoneducation/righton-app/pull/858)_

### CODE REVIEW

File: host/src/lib/HelperFunctions.jsx [hunk 1]
```diff
@@ -206,12 +206,13 @@ export const getShortAnswersPhaseTwo = (shortAnswerResponses, teamsArray, curren
     const correctChoiceIndex = choices.findIndex(choice => choice.isCorrect);
     let answersArray = Array.from({length: choices.length ?? 0}, (item, index) => ({ count: 0, teams: [], isCorrect: index === correctChoiceIndex ? true : false }));
     answers.forEach(({team, answer}) => {
-      choices.forEach((choice, index) => {
-        if (answer.answerContent.rawAnswer === choice.value) {
-          answersArray[index].count += 1;
-          answersArray[index].teams.push({name: team.name});
+      for (let i = 0; i < choices.length; i++){
```
Comment (drewjhart) [inline host/src/lib/HelperFunctions.jsx:209]: breaks the loop after the first match is found, so duplicates aren't created

File: host/src/lib/HelperFunctions.jsx [hunk 2]
```diff
@@ -271,12 +272,22 @@ export const checkEqualityWithPrevAnswer = (normValue, prevAnswerValue, prevAnsw
...
 };
 
-export const checkEqualityWithOtherAnswers = (normValue, normType, prevAnswer) => {
+export const checkEqualityWithOtherAnswers = (rawAnswer, normValue, normType, prevAnswer) => {
   // loop through each of the normalized answers in each of the previous answers
+  console.log(normValue, normType, prevAnswer);
   for (let i = 0; i < prevAnswer.normAnswer.length; i++) {
       if (normType === prevAnswer.normAnswer[i].type && checkEqualityWithPrevAnswer(normValue, prevAnswer.normAnswer[i].value, prevAnswer.normAnswer[i].type)) {
         return true;
       }
```
Comment (drewjhart) [inline host/src/lib/HelperFunctions.jsx:281]: these two checks are last ditch and a bit lazy, but they're trying to capture some edge conditions as best as possible that arose from running tests. More info can be found at the later part of this doc:
https://github.com/rightoneducation/righton-app/wiki/Short-Answer-Response-%E2%80%90-Equality-Checks

File: host/src/pages/GameInProgress.jsx [hunk 3]
```diff
@@ -113,19 +114,26 @@ export default function GameInProgress({
     answerTeams: noResponseTeams,
   };
   // data object used in Victory graph for real-time responses
-  const data = (isShortAnswerEnabled && (currentState === GameSessionState.CHOOSE_CORRECT_ANSWER || currentState === GameSessionState.PHASE_1_DISCUSS))
-    ? buildVictoryDataObjectShortAnswer(
-        shortAnswerResponses, 
-        noResponseObject
-      ) 
+  const data = (isShortAnswerEnabled )
```
Comment (drewjhart) [inline host/src/pages/GameInProgress.jsx:117]: requirement for victory object to also have a phase two state

File: play/src/components/GameSessionSwitch.tsx [hunk 4]
```diff
@@ -50,13 +50,20 @@ export default function GameSessionSwitch({
     (isShortAnswerEnabled
       ? currentQuestion?.responses?.reduce(
           (acc: IChoice[], response: IResponse) => {
-            console.log(response);
-
+            const shouldAddResponse =
```
Comment (drewjhart) [inline play/src/components/GameSessionSwitch.tsx:53]: In Phase Two, we only want to display answers that the teacher has selected via Featured Mistakes or the correct answer. This accomplishes that.

File: play/src/lib/HelperFunctions.tsx [hunk 5]
```diff
@@ -296,8 +296,10 @@ export const handleNormalizeAnswers = (currentContents: any) => { // eslint-disa
           //  reducing case and removing characters
 
           // this extracts numeric values from a string and adds them to the normalized text array.
```
Comment (drewjhart) [inline play/src/lib/HelperFunctions.tsx:298]: Adjustments here are made to handle conditions with %, that arose in testing.

Given two inputs of '5%' and '50%', our original algorithm would extract the 5 and the 50, respectively, leaving '%' and '%', which would then mistakenly produce equality. Now we cut that out and rely on a string comparison later in the code to handle any matching percentages.
